src/main/ruby/core/signal.rb

```diff
@@ -120,6 +120,8 @@ def self.trap(sig, prc=nil, &block)
     old ? old : nil
   end
 
+  NSIG = Numbers.keys.max + 1
```

Comment:
Can you define this higher next to other constants (~line 50)?
Also, it could either compute the max as part of the existing loop, or use `NSIG` from the Rubinius config (would need to be added, I can help with that).

Reply:
Moved this ~line 50.